### PR TITLE
Add HyPhy tools for clade analysis

### PR DESCRIPTION
**Pull request by davebx**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection

This PR and #3598 _should_ each merge fine with or without the other.

- Add tool to annotate a newick tree
- Add tool to replace ambiguous codons in a codon alignment with ---
- Add wrapper for HyPhy Contrast-FEL
- Add tool to translate an in-frame codon alignment to proteins

### CODE REVIEW

File: tools/hyphy/hyphy_annotate.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            <param name="multiple_hits" value="None" />
+            <param name="root" value="REFERENCE" />
+            <param name="label" value="Annotated" />
+            <param name="log" value="false" />
+            <output name="labeled_tree_int" file="annotate-out1-int.nhx" />
+            <output name="labeled_tree_full" file="annotate-out1-full.nhx" />
+            <output name="labeled_tree_clade" file="annotate-out1-clade.nhx" />
+        </test>
+    </tests>
+    <help></help>
```
Comment (bgruening) [inline tools/hyphy/hyphy_annotate.xml:40]: help missing?

File: tools/hyphy/hyphy_conv.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    </outputs>
+    <tests>
+        <test>
+            <param name="input_file" ftype="fasta" value="conv-in1.fa"/>
+            <param name="deletions" value="Keep Deletions" />
+            <output name="conv_output" file="conv-out1.fa" />
+        </test>
+    </tests>
+    <help><![CDATA[
+HyPhy-CONV: Translate an in-frame codon alignment to proteins
```
Comment (bgruening) [inline tools/hyphy/hyphy_conv.xml:35]: some more help missing?

File: tools/hyphy/hyphy_conv.xml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            > '$conv_log'
+        @ERRORS@
+    ]]></command>
+    <inputs>
+        <param name="input_file" type="data" format="fasta" label="Input codon alignment"/>
+        <expand macro="gencode"/>
+        <param name="deletions" type="boolean" truevalue="Keep Deletions" falsevalue="Skip Deletions" label="Keep deletions" help="If this is checked, deletions will be left in the output file"/>
+    </inputs>
+    <outputs>
+        <data name="conv_log" format="txt"/>
```
Comment (bgruening) [inline tools/hyphy/hyphy_conv.xml:24]: optional?

File: tools/hyphy/strike_ambigs.xml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0"?>
+<tool id="strike_ambigs" name="Replace ambiguous codons" version="@VERSION@">
```
Comment (bgruening) [inline tools/hyphy/strike_ambigs.xml:2]: start tool ID with hyphy? and the filename

File: tools/hyphy/strike_ambigs.xml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        <test>
+            <param name="alignment" ftype="fasta" value="strike-ambigs-in1.fa"/>
+            <output name="output" file="strike-ambigs-out1.fa" />
+        </test>
+        <test>
+            <param name="alignment" ftype="fasta" value="strike-ambigs-in2.fa"/>
+            <output name="output" file="strike-ambigs-out2.fa" />
+        </test>
+    </tests>
+    <help></help>
```
Comment (bgruening) [inline tools/hyphy/strike_ambigs.xml:29]: help missing ...

File: tools/hyphy/hyphy_annotate.xml (hunk 1 above)
Comment (davebx) [reply tools/hyphy/hyphy_annotate.xml:40]: Uf, I knew I'd missed something. I'll rewrite the helps.

File: tools/hyphy/test-data/conv-out1.fa [hunk 6]
```diff
@@ -1,7 +1,27 @@
-Universal
-Keep Deletions
-Read 8 sequences with 3819 sites.
-Read:
-8 species:{epi_isl_1041406_hCoV_19_USA_NY_PRL_2021_02_08_05H12_2021, epi_isl_1041403_hCoV_19_USA_NY_PRL_2021_02_08_05H08_2021, gb_MW540268_Organism_Severe_acute_respiratory_syndrome_coronavirus_2_Strain_Name_SARS_CoV_2_human_USA_MA_MASPHL_01380_2020_Segment_null_4, gb_MW467454_Organism_Severe_acute_respiratory_syndrome_coronavirus_2_Strain_Name_SARS_CoV_2_human_EGY_EGY_CCHE57357_A_46_2020_Segment_null_1, gb_MT496989_Organism_Severe_acute_respiratory_syndrome_coronavirus_2_Strain_Name_SARS_CoV_2_human_IND_GBRC63_2020_Segment_null_3, gb_MW525081_Organism_Severe_acute_respiratory_syndrome_coronavirus_2_Strain_Name_SARS_CoV_2_human_USA_MO_CDC_STM_0000025_G03_2021_Segment_null_1, gb_MW518841_Organism_Severe_acute_respiratory_syndrome_coronavirus_2_Strain_Name_SARS_CoV_2_human_USA_CA_CDC_STM_220_2020_Segment_null_1, REFERENCE};
-Total Sites:3819;
```
Comment (bgruening) [inline tools/hyphy/test-data/conv-out1.fa]: Do you know why this changed? And why this file is called .fa?

Comment (davebx) [reply tools/hyphy/test-data/conv-out1.fa]: I'm not actually sure how our changes made this happen, but it should have been .nhx all along, which it now is, and the test output file contains what one would expect with that input data.

File: tools/hyphy/hyphy_conv.xml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        <data name="conv_log" format="txt" from_work_dir="conv.log">
+            <filter>save_log</filter>
+        </data>
+        <data name="proteins" format="nhx" from_work_dir="aa.nhx" />
+    </outputs>
+    <tests>
+        <test>
+            <param name="input_file" ftype="fasta" value="conv-in1.fa"/>
+            <param name="deletions" value="Keep Deletions" />
+            <output name="conv_output" file="conv-out1.nhx" />
```
Comment (bgruening) [inline tools/hyphy/hyphy_conv.xml:34]: @davebx I'm not sure what is going in here. But you are comparing `conv_output` here ... but this output is not defined in this tool. But we have `proteins`?

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Danke Dave!

**Pull request merged**